### PR TITLE
SB-26355 -  Adding cassandra write consistency level to the job.

### PR DESCRIPTION
**Pull request by manjudr**
<table>
<td>
  <a href="https://project-sunbird.atlassian.net/browse/SB-26355" title="SB-26355" target="_blank">StorySB-24793</a>  Data Correction Script 
  </td></table>
  <br />
 

Reading the write consistency config from the job config changes

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by maheshkumargangula — APPROVED** [review]

File: analytics-core/src/main/scala/org/ekstep/analytics/framework/util/CommonUtil.scala [hunk 1]
```diff
@@ -100,7 +100,7 @@ object CommonUtil {
   def getSparkSession(parallelization: Int, appName: String, sparkCassandraConnectionHost: Option[AnyRef] = None,
                       sparkElasticsearchConnectionHost: Option[AnyRef] = None, readConsistencyLevel: Option[String] = None,
                       sparkRedisConnectionHost: Option[AnyRef] = None, sparkRedisDB: Option[AnyRef] = None,
-                      sparkRedisPort: Option[AnyRef] = Option("6379")): SparkSession = {
+                      sparkRedisPort: Option[AnyRef] = Option("6379"), writeConsistencyLevel: Option[String] = None): SparkSession = {
```
Comment (anandp504) [inline analytics-core/src/main/scala/org/ekstep/analytics/framework/util/CommonUtil.scala:103]: Please use a default consistency level in the function parameter.

File: analytics-core/src/main/scala/org/ekstep/analytics/framework/util/CommonUtil.scala [hunk 2]
```diff
@@ -125,6 +125,9 @@ object CommonUtil {
       if (readConsistencyLevel.nonEmpty) {
         conf.set("spark.cassandra.input.consistency.level", readConsistencyLevel.get)
```
Comment (anandp504) [inline analytics-core/src/main/scala/org/ekstep/analytics/framework/util/CommonUtil.scala:126]: Please use getOrElse on an Option

File: analytics-core/src/main/scala/org/ekstep/analytics/framework/util/CommonUtil.scala [hunk 3]
```diff
@@ -125,6 +125,9 @@ object CommonUtil {
...
       }
+      if (writeConsistencyLevel.nonEmpty) {
+        conf.set("spark.cassandra.output.consistency.level", writeConsistencyLevel.get)
```
Comment (anandp504) [inline analytics-core/src/main/scala/org/ekstep/analytics/framework/util/CommonUtil.scala:129]: Please use getOrElse on an Option.

**Review by anandp504 — CHANGES_REQUESTED** [review]

**Review by anandp504 — APPROVED** [review]
Approved

**Pull request merged**